graphql_compiler/macros/macro_edge/helpers.py

```diff
@@ -53,6 +57,214 @@ def get_directives_for_ast(ast):
     return result
 
 
+def get_all_tag_names(ast):
+    """Return a set of strings containing tag names that appear in the query."""
+    return {
+        # Schema validation has ensured this exists
+        directive.arguments[0].value.value
```

Comment:
Are we guaranteed to be able to access `directive.arguments[0].value.value`?

Reply:
yes. There's a comment explaining precisely that.